sections/futures/MarketInfoBox/NextPriceInfoBox.tsx

```diff
@@ -0,0 +1,150 @@
+import { Synths } from '@synthetixio/contracts-interface';
+import Wei, { wei } from '@synthetixio/wei';
+import InfoBox from 'components/InfoBox';
+import PreviewArrow from 'components/PreviewArrow';
+import { FuturesPosition } from 'queries/futures/types';
+import useGetNextPriceDetails from 'queries/futures/useGetNextPriceDetails';
+import React from 'react';
+import { useMemo } from 'react';
+import { useRecoilValue } from 'recoil';
+import { networkState } from 'store/wallet';
+import styled from 'styled-components';
+import { computeNPFee } from 'utils/costCalculations';
+import { formatCurrency, formatPercent, zeroBN } from 'utils/formatters/number';
+import { getMarketKey } from 'utils/futures';
+
+import { PositionSide } from '../types';
+
+type NextPriceInfoBoxProps = {
+	position: FuturesPosition | null;
+	isMarketClosed: boolean;
+	tradeSize: Wei;
+	marketMaxLeverage: Wei | undefined;
+	currencyKey: string;
+	side: PositionSide;
+};
+
+const NextPriceInfoBox: React.FC<NextPriceInfoBoxProps> = ({
+	position,
+	isMarketClosed,
+	marketMaxLeverage,
+	currencyKey,
+	side,
+	tradeSize,
+}) => {
+	const network = useRecoilValue(networkState);
+
+	const totalMargin = position?.remainingMargin ?? zeroBN;
+	const availableMargin = position?.accessibleMargin ?? zeroBN;
+	const buyingPower =
+		position && position?.remainingMargin.gt(zeroBN)
+			? position?.remainingMargin?.mul(marketMaxLeverage ?? zeroBN)
+			: zeroBN;
+
+	const marginUsage =
+		position && position?.remainingMargin.gt(zeroBN)
+			? position?.remainingMargin?.sub(position?.accessibleMargin).div(position?.remainingMargin)
+			: zeroBN;
+
+	const positionSize = position?.position?.size ? wei(position?.position?.size) : zeroBN;
+
+	const nextPriceDetailsQuery = useGetNextPriceDetails(getMarketKey(currencyKey, network.id));
+	const nextPriceDetails = nextPriceDetailsQuery?.data;
+
+	const orderDetails = useMemo(() => {
+		const newSize = side === PositionSide.LONG ? tradeSize : -tradeSize;
+		return { newSize, size: (positionSize ?? zeroBN).add(newSize).abs() };
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [side, positionSize]);
+
+	const { commitDeposit, nextPriceFee } = useMemo(
+		() => computeNPFee(nextPriceDetails, wei(orderDetails.newSize)),
+		[nextPriceDetails, orderDetails]
+	);
+
+	const totalDeposit = useMemo(() => {
+		return (commitDeposit ?? zeroBN).add(nextPriceDetails?.keeperDeposit ?? zeroBN);
+	}, [commitDeposit, nextPriceDetails?.keeperDeposit]);
+
+	// const nextPriceDiscount = useMemo(() => {
+	// 	return (nextPriceFee ?? zeroBN).sub(commitDeposit ?? zeroBN);
+	// }, [commitDeposit, nextPriceFee]);
+
+	const previewTradeData = React.useMemo(() => {
+		const potentialAvailableMargin = availableMargin?.sub(totalDeposit) ?? zeroBN;
+		const potentialBuyingPower =
+			totalMargin.sub(totalDeposit)?.mul(marketMaxLeverage)?.abs() ?? zeroBN;
```

Comment:
This should include the estimated trade as well as the deposit. There is no guarantee this will match exactly with the actual executed trade, however it's more confusing to ignore the trade itself in this calculation.

In this example I'm placing a trade that will take me almost to max leverage, however buying power stays almost the same:
<img width="299" alt="image" src="https://user-images.githubusercontent.com/10401554/172222708-b970b23e-0c68-4ba6-81d0-2ee89b9f6cd5.png">

Reply:
@Tburm if you could try this one again - wondering what the result will be on the latest deployment